File: src/models/employee.cpp
#include "Employee.h"
#include <iostream>
using namespace std;

Employee::Employee(const string& empName, int empId) 
    : name(empName), id(empId) {}

void Employee::displayDetails() const {
        cout << "Name: " << name << "\n"
              << "ID: " << id << "\n"
              << "Salary: " << calculateSalary() << "\n";
}

File: src/models/employee.ts
import { IEmployee } from "../interfaces/iEmployee";

export abstract class Employee implements IEmployee {
    constructor(public name: string, public id: number) {}

    abstract calculateSalary(): number;

    displayDetails(): void {
        console.log(`Name: ${this.name}`);
        console.log(`ID: ${this.id}`);
        console.log(`Salary: ${this.calculateSalary()}`);
    }
}